entrypoint.sh

```diff
@@ -19,25 +19,25 @@ REPORTFILE=${NOW}-perftest-${TEST_SCENARIO}-report.csv
 LOGFILE=${JM_LOGS}/perftest-${TEST_SCENARIO}.log
 
 # Run the test suite
-jmeter -n -t ${SCENARIOFILE} -e -l "${REPORTFILE}" -o ${JM_REPORTS} -j ${LOGFILE} -f -Jenv="${ENVIRONMENT}"
```

Comment:
Just checking that taking `-Jenv="${ENVIRONMENT}"` away doesn't cause any problem when this is deployed within the CDP platform? As I think the env var is automatically set for us depending on the environment we're publishing to.

Reply:
Good question, TDLR; it doesn't cause any issues.

The `ENVIRONMENT` variable on CDP is set to the given environment "stuff" is currently running in, e.g. `perf-test`, `test`, `dev`, etc.

In terms of the example JMeter scenario provided by CDP the `ENVIRONMENT` variable translates to `${ENV}` in the entrypoint script. This is currently inserted into the JMeter scenario via the hostname field in the request e.g. `service-name.${__P(env)}.cdp-int.defra.cloud` (see [here](https://github.com/DEFRA/cdp-perf-test-suite-template/blob/19ee56b59cce050299fbd234d47e66b8943a2ff2/scenarios/test.jmx#L23)).

The downside of the example is that unless the user **always** wants to run the performance tests against CDP they would have to manually edit the scenario file (i.e. if someone wanted to execute a smoke test against a local stack while creating scenarios they would need to change a couple of fields, then remember to revert them before saving, GUI tools FTW).

Also, as of recently, CDP only want/allow performance tests to run on the CDP platform against the `perf-test` environment, so the `${ENV}` variable baked into a CDP specific URL in the example above no longer makes sense. This is also why I've explicitly set the default scenario to run with values of `https` for the protocol, `pha-import-notifications.perf-test.cdp-int.defra.cloud` for the hostname and an empty port. These values can be overridden as per the instructions in the [updated README](https://github.com/DEFRA/pha-import-notifications-perf/pull/1/commits/4dc6d562a736fe14c4834d94f49ad3706a31d7bb) for local testing.

For additional some context, the GUI of JMeter splits URLs by protocol, hostname and port, hence why I've split them out like this for greater flexibility (see screenshot of JMeter GUI). ![image](https://github.com/user-attachments/assets/52df0cf5-2360-4649-8f60-89cd92a0b786).